pybindsrc/module.cpp

```diff
@@ -8,19 +8,19 @@
 
 #include "pybind11/pybind11.h"
 #include "pybind11/stl.h"
-#include "submodules.hpp"
 
 namespace py = pybind11;
 
 namespace dunedaq::nddetdataformats::python {
 
+  extern void register_mpd(pybind11::module &);    
```

Comment:
While we've been using extern register_*, it's not great from code/compilation perspective. I would keep the declaration of the register functions in a separate header, changing the name from submodules.hpp to something more appropriate, like registrators.hpp